TYPO3.Flow/Classes/TYPO3/Flow/Security/Context.php

```diff
@@ -837,7 +864,18 @@ public function getContextHash()
             $this->initialize();
         }
         if ($this->contextHash === null) {
-            $this->contextHash = md5(implode('|', array_keys($this->getRoles())));
+            $contextHashSoFar = implode('|', array_keys($this->getRoles()));
+
+            $this->withoutAuthorizationChecks(function () use (&$contextHashSoFar) {
+                foreach ($this->globalObjects as $globalObjectsRegisteredClassName) {
+                    $globalObject = $this->objectManager->get($globalObjectsRegisteredClassName);
+                    if ($globalObject instanceof CacheAwareInterface) {
+                        $contextHashSoFar .= $globalObject->getCacheEntryIdentifier();
```

Comment:
Cool. Let's just use a different one than we already use for the role identifiers to make super sure ;)